packages/shell-evaluator/src/shell-evaluator.ts

```diff
@@ -35,15 +35,17 @@ class ShellEvaluator<EvaluationResultType = ShellResult> {
   // eslint-disable-next-line complexity
   private async innerEval(originalEval: EvaluationFunction, input: string, context: object, filename: string): Promise<any> {
     const { shellApi } = this.instanceState;
-    const argv = input.trim().replace(/;$/, '').split(/\s+/g);
+    const trimmedInput = input.trim();
+    const argv = trimmedInput.replace(/;$/, '').split(/\s+/g);
     const cmd = argv.shift() as keyof typeof shellApi;
 
     if (
       shellApi[cmd]?.isDirectShellCommand &&
       shellApi[cmd]?.acceptsRawInput &&
       !(argv[0] ?? '').startsWith('(')
     ) {
-      return shellApi[cmd](input);
+      const code = /^(\S+)$/.test(trimmedInput) ? '' : trimmedInput.replace(/^\S+\s+/, '');
```

Comment:
This is the same as

```suggestion
      const code = trimmedInput.replace(/^\S+\s*/, '');
```

right?

Also, I wouldn't call this `code`, it's code when it refers to the `edit` command but this is fairly generic, I would call it something like `rawArgument` or `rawInput` (also because that's consistent with `acceptsRawInput` elsewhere)